sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -585,7 +585,7 @@ abstract class StreamExecution(
       options: Map[String, String],
       inputPlan: LogicalPlan): StreamingWrite = {
     val writeBuilder = table.newWriteBuilder(new DataSourceOptions(options.asJava))
-      .withQueryId(runId.toString)
+      .withQueryId(id.toString)
```

Comment:
this is an unrelated change, but it was obviously a mistake: the sink doesn't care about the `runId` which gets changed after query restart. The sink needs the `id` which is reliable across the life cycle of the stream query.

No builtin streaming sinks use this id, so this is for future-proof.